Purchases/Purchasing/Purchases.swift

```diff
@@ -1160,6 +1160,43 @@ public extension Purchases {
     }
 #endif
 
+    /**
+     * Use this function to retrieve a `Bool` if eligible for a given `StoreProductDiscount` for a given `StoreProduct`.
+     *
+     * - Parameter discount: The `StoreProductDiscount` to apply to the product.
+     * - Parameter product: The `StoreProduct` the user intends to purchase.
+     * - Parameter completion: A completion block that is called when the `Bool` is returned.
+     * If it was not successful, there will be an `Error`.
+     */
+    @available(iOS 12.2, macOS 10.14.4, macCatalyst 13.0, tvOS 12.2, watchOS 6.2, *)
+    @objc(checkEligiblilityForPromotionalOffer:product:completion:)
+    func checkEligiblilityForPromotionalOffer(forProductDiscount discount: StoreProductDiscount,
+                                              product: StoreProduct,
+                                              completion: @escaping (Bool, Error?) -> Void) {
+        if let sk1Product = product.sk1Product {
+            purchasesOrchestrator.promotionalOffer(forProductDiscount: discount,
+                                                   product: sk1Product) { promotionalOffer, error in
+                completion(promotionalOffer != nil, error)
+            }
+        } else {
+            fatalError("StoreKit2 not supported yet")
+        }
+    }
+
+    /**
+     * Use this function to retrieve a `Bool` if eligible for a given `StoreProductDiscount` for a given `StoreProduct`.
```

Comment:
wonder if we should add something about how this is different from the existing function (intro only)... maybe both should have a comment differentiating from the other...

Reply:
I think that would be a good idea since these two different discounts confuse me 😛 But I don't know if I know how to write that verbiage. Might need a volunteer here 😇

Reply:
how about: 
```swift
/** Use this method to find eligibility for this user for 
* [iOS Promotional Offers](https://docs.revenuecat.com/docs/ios-subscription-offers).
* - Note: If you're looking to use free trials or Introductory Offers instead,
* use ``Purchases/checkTrialOrIntroductoryPriceEligibility``. 
```
And then a similar thing on the other one

Reply:
Alright, updated if you want to take look again whenever you get a chance! 😊